scout_utils.c

```diff
@@ -220,16 +226,21 @@ const char *unique_resource_id(const char *scout_wrapper_type, zval *resource_id
         return "";
     }
 
-    DYNAMIC_MALLOC_SPRINTF(ret, len,
 #if PHP_VERSION_ID >= 80100
+    DYNAMIC_MALLOC_SPRINTF(ret, len,
         "%s_handle(%ld)_type(%d)",
+        scout_wrapper_type,
+        Z_RES_HANDLE_P(resource_id),
+        Z_RES_TYPE_P(resource_id)
+    );
 #else
+    DYNAMIC_MALLOC_SPRINTF(ret, len,
         "%s_handle(%d)_type(%d)",
-#endif
```

Comment:
Indeed, this is not supported by MSVC; however, the [new preprocessor](https://docs.microsoft.com/en-us/cpp/preprocessor/preprocessor-experimental-overview?view=msvc-170) at least has a better error message:

> warning C5101: use of preprocessor directive in function-like macro argument list is undefined behavior